### PR TITLE
Add an api endpoint to get the current version

### PR DESCRIPTION
**Pull request by ameade**
## Description
Add /api/version endpoint that returns {"version": "x.x.x"}

Currently, clients must hardcode the version since it is required via the 'Homebrewery-Version' header for every API request. This change allows a client to inspect the server version before attempting requests.

## Related Issues or Discussions

> [!CAUTION]
> If no issue exists yet, create it, and get agreement on the approach (or paste in a previous agreement from chat, etc.) before moving forward. (Experimental PRs are OK without prior discussion, but do not expect to get merged.)

- Closes #

## QA Instructions, Screenshots, Recordings

curl -s http://localhost:8000/api/version

### Reviewer Checklist

*Reviewers, refer to this list when testing features, or suggest new items *
- [ ] Verify new features are functional
  - [ ] Feature A does X
  - [ ] Feature B does Y
- [ ] Verify old features have not broken
  - [ ] Feature Z can still be used
- [ ] Test for edge cases / try to break things
  - [ ] Feature A handles negative numbers
- [ ] Identify opportunities for simplification and refactoring
- [ ] Check for code legibility and appropriate comments

### CODE REVIEW

File: server/version.api.spec.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import packageJSON from '../package.json' with { type: 'json' };
+
+describe('Version API', () => {
+	let app;
+
+	beforeEach(() => {
+		app = express();
+		
+		// Add the version endpoint (same as in app.js)
+		app.get('/api/version', (req, res) => {
```
Comment (G-Ambatte) [inline server/version.api.spec.js:12]: This is required because the test is not loading `server/app.js`, so the actual function is not being tested.
I believe the correct implementation should be to `import app from '../server/app.js';` (or whatever the correct file path is).

File: server/version.api.spec.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	describe('GET /api/version', () => {
+		it('should return the current package version as JSON', async () => {
+			const response = await request(app)
+				.get('/api/version')
+				.expect(200);
+
+			expect(response.body).toEqual({ version: packageJSON.version });
+		});
+
+		it('should return JSON content type', async () => {
```
Comment (G-Ambatte) [inline server/version.api.spec.js:26]: I'm not sure that there's any benefit to running the same function multiple times to look at a different part of the result each time.
That said, I'm not overly upset by it.

**Comment by 5e-Cleric** [comment]
I have a question, is this api meant to be for preventing falied requests, or for external calls to our service?

**Comment by 5e-Cleric** [comment]
retagging @ameade to get the answer, otherwise will be closing this. One week should suffice.

**Comment by 5e-Cleric** [comment]
No response in 2.5 months, closing this.
